gammapy/estimators/points/profile.py

```diff
@@ -103,6 +111,14 @@ def __init__(self, regions, spectral_model=None, **kwargs):
             spectral_model = PowerLawSpectralModel()
 
         self.spectral_model = spectral_model
+
+        if "reoptimize" in kwargs:
+            if kwargs["reoptimize"] is True:
+                raise ValueError(
+                    f"reoptimize=True is not available for {self}. Set to `False` instead."
+                )
+        else:
+            kwargs["reoptimize"] = False
```

Comment:
```suggestion
        reoptimize = kwargs.get("reoptimize", False)
        if reoptimize is True:
                raise ValueError(
                    f"reoptimize=True is not available for {self}. Set to `False` instead."
                )
```
this might be simpler